packages/react-components/src/components/VideoGallery/PinnedParticipantsLayout.tsx

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { LayerHost, mergeStyles, Stack } from '@fluentui/react';
+import { useId } from '@fluentui/react-hooks';
+import React from 'react';
+import { useTheme } from '../../theming';
+import { VideoGalleryRemoteParticipant } from '../../types';
+import { GridLayout } from '../GridLayout';
+import {
+  localVideoTileContainerStyle,
+  localVideoTileWithControlsContainerStyle,
+  LOCAL_VIDEO_TILE_ZINDEX
+} from './styles/FloatingLocalVideo.styles';
+import { isNarrowWidth } from '../utils/responsive';
+import { VideoGalleryStyles } from '../VideoGallery';
+import { FloatingLocalVideo } from './FloatingLocalVideo';
+import { innerLayoutStyle, layerHostStyle, rootLayoutStyle } from './styles/FloatingLocalVideoLayout.styles';
+import { VideoGalleryResponsiveHorizontalGallery } from './VideoGalleryResponsiveHorizontalGallery';
+import { usePinnedParticipantLayout } from './utils/videoGalleryLayoutUtils';
+
+/**
+ * Props for {@link PinnedParticipantsLayout}.
+ *
+ * @private
+ */
+export interface PinnedParticipantsLayoutProps {
+  /**
```

Comment:
A lot of common props shared across layouts can be extracted into a single interface and used here to extend PinnedParticipantsLayoutProps.

Reply:
Great observation. Refactored.